front/lib/api/assistant/pubsub.ts

```diff
@@ -156,24 +156,33 @@ export async function* getMessagesEvents(
     | AgentGenerationSuccessEvent;
 }> {
   const pubsubChannel = getMessageChannelId(messageId);
-  const client = await redisClient();
-  const events = await client.xRead(
-    { key: pubsubChannel, id: lastEventId ? lastEventId : "0-0" },
-    { COUNT: 1, BLOCK: 60 * 1000 }
-  );
-  if (!events) {
-    return;
-  }
-  for (const event of events) {
-    for (const message of event.messages) {
-      const payloadStr = message.message["payload"];
-      const messageId = message.id;
-      const payload = JSON.parse(payloadStr);
-      yield {
-        eventId: messageId,
-        data: payload,
-      };
+  const redis = await redisClient();
+
+  try {
+    while (true) {
+      const events = await redis.xRead(
+        { key: pubsubChannel, id: lastEventId ? lastEventId : "0-0" },
+        // weird, xread does not return on new message when count is = 1. Anything over 1 works.
```

Comment:
wut?

Reply:
This looks like your old comment?

Reply:
Old comment. Will clean up.

Reply:
Why is there a much bigger diff in one case vs the other ?

Reply:
I had updated the first one but forgot to update this one. That's it. This code was never really tested outside of outputting one comment. I am pretty confident with this PR, I tested it pretty well. Except infinite loop on error, but I will take care of it in another PR.